test/integration/framework/etcd.go

```diff
@@ -17,93 +17,102 @@ limitations under the License.
 package framework
 
 import (
+	"context"
 	"fmt"
-	"hash/adler32"
-	"io"
 	"io/ioutil"
 	"math/rand"
+	"net"
 	"os"
 	"os/exec"
-	"path/filepath"
-	"sync"
+	"strings"
 
 	"github.com/golang/glog"
 
 	"k8s.io/kubernetes/pkg/util/env"
 )
 
-var (
-	etcdSetup sync.Once
-	etcdURL   = ""
-)
-
-func setupETCD() {
-	etcdSetup.Do(func() {
-		if os.Getenv("RUNFILES_DIR") == "" {
```

Comment:
`RUNFILES_DIR` is used when this is run under bazel. This should probably be documented better (at least have a comment), but there's unfortunately not really a better way to pass the path of this dependency, since this is implemented in a library that's used by a bunch of individual go tests. I'd rather not break this functionality.

Reply:
Thank you, I had no idea about `RUNFILES_DIR`. I'll fix this PR to not break that. Testing it with this command.
```
 bazel test //test/integration/scheduler:go_default_test --test_arg=-test.v --test_arg=-test.run=xxx --test_output=all
```